cpp/src/arrow/compute/exec/key_encode.cc

```diff
@@ -864,6 +864,17 @@ void KeyEncoder::EncoderBinaryPair::Encode(uint32_t offset_within_row, KeyRowArr
     EncodeImp_fn[dispatch_const](num_processed, offset_within_row, rows, col_prep[0],
                                  col_prep[1]);
   }
+
+  DCHECK(temp1->metadata().is_fixed_length);
+  DCHECK(temp1->length() * temp1->metadata().fixed_length >=
```

Comment:
Please use the more specific macros for DCHECK and ARROW_CHECK statements
```suggestion
  DCHECK_GE(temp1->length() * temp1->metadata().fixed_length,
```
There is a lint check against this, but it occasionally fails to parse in the presence of `operator->`

Reply:
done